tests/form.js

```diff
@@ -527,7 +527,61 @@ define([
                     }
                 });
             });
+        });
+
+        describe("hashtag logic", function () {
+            var form;
+            before(function () {
+                form = util.loadXML("");
+            });
+
+            _.each({
+                "": null,
+                "#": null,
+
+                "#form": null,
+                /* TODO should these pass? (they don't)
+                "#form/": "is",
+                "#form/prop": "has",
```

Comment:
I added these tests to verify things that previously worked continued to work after refactoring and happened to uncover these surprises. I think changing/fixing that is out of scope for this PR.